### DISCUSSION TITLE
C# 7.3 (by .Net 4.8)

### DISCUSSION THREAD

**Discussion by giesalabs** [dahall/TaskScheduler#908] [Q&A] [answered]
I'm try to make the code compatible to C# 7.3 and I see some incompatibility of the code with "null coalescing".

I didn't find on Google the expanded version of a code like this:
https://github.com/dahall/TaskScheduler/blob/270ba8f3553d003d0bf9776657d7ff5b461f0786/TaskService/Task.cs#L3568
Could you help me to expand the line above to make it compatible with older C# 7.3?

**Comment by dahall** [comment, marked as answer]
```cs
public IdleSettings IdleSettings
{
   get
   {
      if (idleSettings is null)
         idleSettings = v2Settings != null ? new IdleSettings(v2Settings.IdleSettings) : new IdleSettings(v1Task);
   }
}
```
